data_updates/Python/iati_rhfp.py

```diff
@@ -0,0 +1,212 @@
+total_data['Code type'] = "Reproductive health care and family planning"
+
+# Groupby country
+
+data = data.groupby(["Year","recipient_name","country","Code type"]).agg({"disbursed":"sum"}).reset_index()
+
+data = data.append(pd.DataFrame(data = total_data),ignore_index=True)
+
+# Put it in millions
+
+data["disbursed"] = data["disbursed"]/1000000
+
+data = data.pivot(index=['Code type','country','recipient_name'], columns='Year', values='disbursed').reset_index()
+
+data["Reporting Organisation Narrative"] = data["country"]
+
+data = data[data['country'] != "Germany"]
+
+data.to_csv("csv/IATI-RHFP-data-v2.csv", encoding='utf-8', index=False)
+
+# Dataset 3
+
+# data = pd.read_csv("https://ddw.devinit.org/api/export/1231")
+
+# data.columns = ["Reporting Organisation Reference","Reporting Organisation Narrative","year","Sector Code - Calculated"
+# , "DAC3 Sector Code - Calculated","Reporting Organisation Type Code","Reporting Organisation Type Name - Calculated"
+# ,"Recipient Code","Recipient Name","Flow Type Code - Calculated","aid_type_di_name","x_transaction_value_usd_m_Sum"]
+
+# data = data[data['Reporting Organisation Reference'].isin(donors_selected['reporting_org_ref'])]
+
+# data = pd.DataFrame(data)
+
+# data = data[(data['Sector Code - Calculated'] != 13010) & (data['Sector Code - Calculated'] != 13081)]
+
+# # Sector code mapping
+
+# data.loc[data['Sector Code - Calculated']==13020,'Sector Code - Calculated'] = "Reproductive health care"
+# data.loc[data['Sector Code - Calculated']==13030,'Sector Code - Calculated'] = "Family planning"
+
+# # Channel of delivery mapping
+
+# data.loc[data['Transaction Receiver Organisation Type'].isin([21,22,23,24]),'Transaction Receiver Organisation Type'] = "NGOs and Civil Society"
+# data.loc[data['Transaction Receiver Organisation Type'].isin([40]),'Transaction Receiver Organisation Type'] = "Multilateral"
+# data.loc[data['Transaction Receiver Organisation Type'].isin([10]),'Transaction Receiver Organisation Type'] = "Public Sector"
+# data.loc[data['Transaction Receiver Organisation Type'].isin([70,73]),'Transaction Receiver Organisation Type'] = "Private Sector"
+# data.loc[data['Transaction Receiver Organisation Type'].isin([80]),'Transaction Receiver Organisation Type'] = "Universities and Research Institutes"
+# data.loc[data['Transaction Receiver Organisation Type'].isin([90]),'Transaction Receiver Organisation Type'] = "Other"
+
+# data = data.groupby(['Reporting Organisation Narrative', 'year','Sector Code - Calculated','Transaction Receiver Organisation Type'])[["x_transaction_value_usd_m_Sum"]].sum().reset_index()
+
+# print(data)
+
+# data.columns = ["Year","Donor Name","Purpose Name","Channel","x_transaction_value_usd_m_Sum"]
+
+# print(data)
+
+# # Put it in millions
+
+# data["x_transaction_value_usd_m_Sum"] = data["x_transaction_value_usd_m_Sum"]/1000000
+
+# data = data[data['country'] != "Germany"]
+
+# data.to_csv("csv/iati_rhfp3.csv", encoding='utf-8', index=False)
```

Comment:
Any chance we can get rid of this large section of commented out code, or will it be useful at some point?

Reply:
It was for a CSV we stopped using so yes. Deleted.